drivers/abp2/abp2.c

```diff
@@ -0,0 +1,227 @@
+#include "ztimer.h"
+
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+#define DEV (dev->params.spi)
+
+#define ABP2_CMD_MEAS       (0xAA)          /**< Start a measurement. */
+#define ABP2_CMD_NOP        (0xF0)          /**< NOP command. */
+
+#define ABP2_RX_BUF_LEN     (8)             /**< Length of the receive buffer. */
+
+
+/** Data to send in order to start a measurement. */
+static const uint8_t data_tx_meas_start[] = { ABP2_CMD_MEAS, 0x00, 0x00 };
+
+/** Data to send in order to start a measurement and read the data of the previous measurement. */
+static const uint8_t data_tx_meas_read[] = { ABP2_CMD_MEAS, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };
+
+/** Data to send in order to read the data of the previous measurement. */
+static const uint8_t data_tx_nop_read[] = { ABP2_CMD_NOP, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };
+
+static const int32_t cntPMax = 15099494;    /**< Output at maximum pressure (counts). */
+static const int32_t cntPMin = 1677722;     /**< Output at minimum pressure (counts). */
+
+static float pOffset = 0.0f;
+
+
+int abp2_init(abp2_t *dev, const abp2_params_t *params)
+{
+    uint8_t status = 0xFF;      /* sensor status byte */
+    int res = ABP2_OK;
+    int count = 10;
+
+    assert(dev && params);
+    dev->params = params;
+
+#if defined(MODULE_ABP2_SPI)
+    res = spi_init_cs(params->spi, params->cs);
+#else
+#pragma message("implement I2C code here")
+#endif
+    if (res) {
+        return ABP2_ERRBUS;
+    }
+
+    /* test status byte: the busy flag should clear in about 5ms */
+    while ((status & ABP2_STATUS_BUSY) && count) {
+#if defined(MODULE_ABP2_SPI)
+        spi_acquire(params->spi, params->cs, SPI_MODE_0, params->clk);
+        status = spi_transfer_byte(params->spi, params->cs, false, ABP2_CMD_NOP);
+        spi_release(params->spi);
+#else
+#pragma message("implement I2C code here")
+#endif
+        ztimer_sleep(ZTIMER_MSEC, 1);
+        count--;
+    }
+    /* the busy flag should be clear */
+    if (status & ABP2_STATUS_BUSY) {
+        res = ABP2_ERRSENS;
+    }
+
+    return res;
+}
+
+int abp2_measure(const abp2_t *dev)
+{
+    uint8_t status = 0xFF;      /* sensor status byte */
+    int res = ABP2_OK;
+    int len = sizeof(data_tx_meas_start) / sizeof(data_tx_meas_start[0]);
+    uint8_t data_rx[ABP2_RX_BUF_LEN];
+
+#if defined(MODULE_ABP2_SPI)
+    res = spi_init_cs(dev->params->spi, dev->params->cs);
+#else
+#pragma message("implement I2C code here")
+#endif
+    if (res) {
+        return ABP2_ERRBUS;
+    }
+#if defined(MODULE_ABP2_SPI)
+    spi_acquire(dev->params->spi, dev->params->cs, SPI_MODE_0, dev->params->clk);
+    spi_transfer_bytes(dev->params->spi, dev->params->cs, false, data_tx_meas_start, data_rx, len);
+    spi_release(dev->params->spi);
+#else
+#pragma message("implement I2C code here")
+#endif
+    status = data_rx[0];
+    if (status & ABP2_STATUS_MASK) {
+        return ABP2_ERRSENS;
+    }
+
+    return res;
+}
+
+uint8_t abp2_getstatus(const abp2_t *dev)
```

Comment:
No, it shouldn't. It's part of one of the 2 methods to read the sensor.

Reply:
Well it depends. In the light of your other comments about not exposing the raw data to the user, I could make 2 versions of `abp2_measure_read()`:
- one would be non-blocking, i.e. the current version
- one would be blocking: it would merge `abp2_measure()`, `abp2_getstatus()` and `abp2_read_raw()`.

In that case, how should I name the blocking and non-blocking versions of the same function ?

Reply:
How would the use-case for the non-blocking version look like?

Reply:
Considering that RIOT supports multi-threading, maybe there isn't any. In that context, a blocking function is not an issue, is it ?

Reply:
I changed the code so that the user can call either of these 2 functions:
-  `int abp2_read(const abp2_t *dev, int32_t *press, int32_t *temp)`: blocking function
-  `int abp2_read_nb(const abp2_t *dev, int32_t *press, int32_t *temp)`: non-blocking function

I left the lower level functions available to the advanced user who may not need the conversion to physical quantities, or who needs to diagnose a faulty sensor.